dll/win32/browseui/shellfind/CSearchBar.cpp

```diff
@@ -131,6 +137,34 @@ LRESULT CSearchBar::OnSearchButtonClicked(WORD wNotifyCode, WORD wID, HWND hWndC
         return 0;
     }
 
+    // See if we have a szFileName, our first character is not an asterisk,
+    // and we have room in MAX_PATH to add an asterisk at the beginning of szFileName then add it
+    if(wcscmp(pSearchStart->szFileName, L"") != 0)
+        {
+        len = wcslen(pSearchStart->szFileName);
+        if(len > 0)
+            {
+            ptrchar = pSearchStart->szFileName[len - 1];
+            memcpy(endchar, &ptrchar, 2);
+            ptrchar = pSearchStart->szFileName[0];
+            memcpy(startchar, &ptrchar, 2);
```

Comment:
You could have just done `startchar = *ptrchar;` if the variable "startchar" was `WCHAR startchar;`